app/assets/javascripts/pageflow/editor/api/file_types.js

```diff
@@ -29,26 +29,42 @@ pageflow.FileTypes = pageflow.Object.extend({
   },
 
   setup: function(serverSideConfigs) {
+    var clientSideConfigs = this.clientSideConfigs;
     this._setup = true;
 
-    this.fileTypes = _.map(serverSideConfigs, function(serverSideConfig) {
-      var clientSideConfig = this.clientSideConfigs[serverSideConfig.collectionName];
-
-      if (!clientSideConfig) {
-        throw 'Missing client side config for file type "' + serverSideConfig.collectionName + '"';
+    this.fileTypes = new pageflow.
+      FileTypesCollection(_.map(serverSideConfigs, function(serverSideConfig) {
+        var clientSideConfig = clientSideConfigs[serverSideConfig.collectionName];
+
+        if (!clientSideConfig) {
+          throw 'Missing client side config for file type "' +
+            serverSideConfig.collectionName + '"';
+        }
+
+        _(this.clientSideConfigModifications[serverSideConfig.collectionName])
+          .each(function(modification) {
+            this.lintModification(modification, serverSideConfig.collectionName);
+            this.applyModification(clientSideConfig, modification);
+          }, this);
+
+        return new pageflow.FileType(_.extend({}, serverSideConfig, clientSideConfig));
+      }, this));
+
+    _.map(this.fileTypes.fileTypes, function(fileType) {
+      if(fileType && fileType.nestedFileTypes.fileTypes){
+        fileType.nestedFileTypes = new pageflow.FileTypesCollection(
+          _.map(fileType.nestedFileTypes.fileTypes, function(nestedFileType) {
```

Comment:
This relies on parsing string representations of the nested files types given in the client side configuration into a `FileTypesCollection` which is then overwritten by a new `FileTypesCollection`. This change of types seems overly confusing to me. 

Moreover, I'd suggest including the names of the nested file types in the config seed data generated by the server. That way there is no need to specify the nested file types again when registering the parent file type in the JS code.